### PR TITLE
Fix argument must be of type array

### PR DESCRIPTION
**Pull request by aschempp**
https://terminal42.sentry.io/share/issue/a38e9e81a9e8410b927ad9caa9e420a3/

### CODE REVIEW

File: core-bundle/src/Resources/contao/classes/DataContainer.php [hunk 1]
```diff
@@ -449,7 +449,7 @@ protected function row($strPalette=null)
 			$key = (Input::get('act') == 'editAll') ? 'FORM_FIELDS_' . $suffix : 'FORM_FIELDS';
 
 			// Calculate the current palette
-			$postPaletteFields = implode(',', Input::post($key));
+			$postPaletteFields = implode(',', (array) Input::post($key));
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/classes/DataContainer.php:452]: Why would we want to catch this error? It only occurs if the user input is invalid, so the following seems more appropriate to me:

```php
if (!\is_array($post = Input::post($key)) {
    throw new \RuntimeException('Invalid data submitted!');
}

$postPaletteFields = implode(',', $post);
```

WDYT?

Comment (aschempp) [reply core-bundle/src/Resources/contao/classes/DataContainer.php:452]: Idk, maybe one of the selected records did not contain any field in the palette?

Comment (leofeyer) [reply core-bundle/src/Resources/contao/classes/DataContainer.php:452]: Can you debug this?

Comment (aschempp) [reply core-bundle/src/Resources/contao/classes/DataContainer.php:452]: I don't think I can, this just randomly appeared in Sentry, I have no idea what users did. But splitting a value that you're not sure is a string is definitly wrong (Rector for example would add a string casting here as well).

Comment (fritzmg) [reply core-bundle/src/Resources/contao/classes/DataContainer.php:452]: Well what did the stack trace reported from Sentry look like?

Comment (aschempp) [reply core-bundle/src/Resources/contao/classes/DataContainer.php:452]: The sentry report (including stack trace) is linked in the initial comment 😉

Comment (fritzmg) [reply core-bundle/src/Resources/contao/classes/DataContainer.php:452]: Yes, but the public report does not show the POST data.

Comment (aschempp) [reply core-bundle/src/Resources/contao/classes/DataContainer.php:452]: I can't see any post data in the private view either. But it obviously was not an array, maybe you can explain what yuo're looking for?

Comment (fritzmg) [reply core-bundle/src/Resources/contao/classes/DataContainer.php:452]: > maybe you can explain what yuo're looking for?

Just some clues on how to reproduce this.

**Pull request closed without merging**